react-blocks/src/blocks/Counter/CounterBlock.js

```diff
@@ -0,0 +1,155 @@
+import { CounterIcon } from './CounterIcon.js';
+import { Counter } from './Counter.js';
+
+export class CounterBlock {
+  constructor() {
+    const { registerBlockType } = wp.blocks;
+    const { withSelect } = wp.data;
+
+    registerBlockType( 'planet4-blocks/counter', {
+      title: 'Counter',
+      icon: CounterIcon,
+      category: 'planet4-blocks',
+
+      // Transform the shortcode into a Gutenberg block
+      // this is used when a user clicks "Convert to blocks"
+      // on the "Classic Editor" block
+      transforms: {
+        from: [
+          {
+            type: 'shortcode',
+            // Shortcode tag can also be an array of shortcode aliases
+            tag: 'shortcake_counter',
+            attributes: {
+              style: {
+                type: 'string',
+                shortcode: ( { named: { style = 'plain' } } ) => style,
+              },
+              title: {
+                type: 'string',
+                shortcode: ( { named: { title = '' } } ) => title,
+              },
+              description: {
+                type: 'string',
+                shortcode: ( { named: { description = '' } } ) => description,
+              },
+            },
+          },
+        ]
+      },
+      attributes: {
+        title: {
+          type: 'string',
+        },
+        description: {
+          type: 'string',
+        },
+        style: {
+          type: 'string',
+          default: 'plain'
+        },
+        completed: {
+          type: 'integer',
+        },
+        completed_api: {
+          type: 'string',
+        },
+        target: {
+          type: 'integer',
+        },
+        text: {
+          type: 'string',
+        }
+      },
+      // withSelect is a "Higher Order Component", it works as
+      // a Decorator, it will provide some basic API functionality
+      // through `select`.
+      edit: withSelect( ( select ) => {
+        const tagsTaxonomy = 'post_tag';
```

Comment:
these lines (68-80) are not needed for this block

Reply:
I agree with @kirdia , The `withSelect()` is a HOC which use to fetch data from Redux. at here, we don't need it so you can simply use arrow function like below and put all stuff into it.
```
edit: ({ isSelected, attributes, setAttributes }) => {
    // edit function stuff.
},
```